lib/mkmf-rice.rb

```diff
@@ -1,93 +1,8 @@
-  end
-
-  MakeMakefile::CONFTEST_C = "#{CONFTEST}.cc"
-
-  MakeMakefile.module_eval do
-    CONFTEST_C = "#{CONFTEST}.cc"
-    def cc_config(opt="")
-      conf = RbConfig::CONFIG.merge('hdrdir' => $hdrdir.quote, 'srcdir' => $srcdir.quote,
-                                    'arch_hdrdir' => $arch_hdrdir.quote,
-                                    'top_srcdir' => $top_srcdir.quote)
-      conf
-    end
-
-    def link_config(ldflags, opt="", libpath=$DEFLIBPATH|$LIBPATH)
-      librubyarg = $extmk ? $LIBRUBYARG_STATIC : "$(LIBRUBYARG)"
-      conf = RbConfig::CONFIG.merge('hdrdir' => $hdrdir.quote,
-                                    'src' => "#{conftest_source}",
-                                    'arch_hdrdir' => $arch_hdrdir.quote,
-                                    'top_srcdir' => $top_srcdir.quote,
-                                    'INCFLAGS' => "#$INCFLAGS",
-                                    'CPPFLAGS' => "#$CPPFLAGS",
-                                    'CFLAGS' => "#$CFLAGS",
-                                    'ARCH_FLAG' => "#$ARCH_FLAG",
-                                    'LDFLAGS' => "#$LDFLAGS #{ldflags}",
-                                    'LOCAL_LIBS' => "#$LOCAL_LIBS #$libs",
-                                    'LIBS' => "#{librubyarg} #{opt} #$LIBS")
-      conf['LIBPATH'] = libpathflag(libpath.map {|s| RbConfig::expand(s.dup, conf)})
-      conf
-    end
-
-    @lang = Hash.new(self)
-
-    def self.[](name)
-      @lang.fetch(name)
-    end
-
-    def self.[]=(name, mod)
-      @lang[name] = mod
-    end
-
-    MakeMakefile["C++"] = Module.new do
-      include MakeMakefile
-      extend self
-
-      CONFTEST_CXX = "#{CONFTEST}.#{config_string('CXX_EXT') || CXX_EXT[0]}"
-
-      TRY_LINK_CXX = config_string('TRY_LINK_CXX') ||
-        ((cmd = TRY_LINK.gsub(/\$\(C(?:C|(FLAGS))\)/, '$(CXX\1)')) != TRY_LINK && cmd) ||
-        "$(CXX) #{OUTFLAG}#{CONFTEST}#{$EXEEXT} $(INCFLAGS) $(CPPFLAGS) " \
-                     "$(CXXFLAGS) $(src) $(LIBPATH) $(LDFLAGS) $(ARCH_FLAG) $(LOCAL_LIBS) $(LIBS)"
-
-      def have_devel?
-        unless defined? @have_devel
-          @have_devel = true
-          @have_devel = try_link(MAIN_DOES_NOTHING)
-        end
-        @have_devel
-      end
-
-      def conftest_source
-        CONFTEST_CXX
-      end
-
-      def cc_command(opt="")
-        conf = cc_config(opt)
-        RbConfig::expand("$(CXX) #$INCFLAGS #$CPPFLAGS #$CXXFLAGS #$ARCH_FLAG #{opt} -c #{CONFTEST_CXX}",
-                         conf)
-      end
-
-      def link_command(ldflags, *opts)
-        conf = link_config(ldflags, *opts)
-        RbConfig::expand(TRY_LINK_CXX.dup, conf)
-      end
-    end
-  end
-end
+IS_MSWIN = RbConfig::CONFIG['host_os'].match?(/mswin/)
+IS_MINGW = RbConfig::CONFIG['host_os'].match?(/mingw/)
+IS_DARWIN = RbConfig::CONFIG['host_os'].match?(/darwin/)
 
 # The cpp_command is not overwritten in the experimental mkmf C++ support.
 # See https://bugs.ruby-lang.org/issues/17578
```

Comment:
If this still an issue, we should comment on my original bug and see if we can get this fixed.